akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -204,6 +204,8 @@ import scala.util.control.NonFatal
       } else if (downstreamCanceled) {
         upstreamCompleted = true
         tryCancel(subscription)
+      } else if (upstream != null) { // reactive streams spec 2.5
+        tryCancel(subscription)
```

Comment:
nice simple fix; good

Reply:
Should add TCK for every kind of Source ,Flow and Sink?

Reply:
We have... also, since akka streams is interpreted it does not make much of a difference - since the rules are protected by the interpreter. We know what we’re doing here